Makefile

```diff
@@ -1,7 +1,199 @@
-all:
-	$(MAKE) -C src $@
+.POSIX:
 
-%: FORCE
-	$(MAKE) -C src $@
+CXX = c++
 
-.PHONY: all FORCE
+debug = no
+static = no
+gzip_man = yes
+# to get format compatible with GitHub archive use "gzip -S .gz" here
+compress_bin = bzip
+
+compress-suffix-bzip = bz2
+compress-suffix-zstd = zst
+
+CPPFLAGS-debug-yes = -DKAK_DEBUG
+CXXFLAGS-debug-yes = -O0 -g
+suffix-debug-yes = .debug
+
+CXXFLAGS-debug-no = -O3
+suffix-debug-no = .opt
+
+CXXFLAGS-sanitize-address = -fsanitize=address
+LDFLAGS-sanitize-address = -lasan
+suffix-sanitize-address = .san_a
+
+CXXFLAGS-sanitize-undefined = -fsanitize=undefined
+
+LDFLAGS-sanitize-undefined = -lubsan
+suffix-sanitize-undefined = .san_u
+
+LDFLAGS-static-yes = -static -pthread
+
+version != cat .version 2>/dev/null || git describe --tags HEAD 2>/dev/null || echo unknown
+
+sources != find src -type f -name '*.cc' | sed -e '/\.version\.cc/d'
+deps != touch src/main.d && find src -type f -name '*.d'
```

Comment:
Why do we touch main.d here, would not that always trigger recompilation of main.cc ?